core/src/Streamly/Internal/Unicode/Stream.hs

```diff
@@ -1103,12 +1097,12 @@ isSpace c
 -- >>>  Stream.toList $ words Fold.toList (Stream.fromList "fold these     words")
 -- ["fold","these","words"]
 --
--- > words = S.wordsBy isSpace
+-- > words = Stream.wordsBy isSpace
 --
 -- /Pre-release/
 {-# INLINE words #-}
-words :: (Monad m, IsStream t) => Fold m Char b -> t m Char -> t m b
-words = S.wordsBy isSpace
+words :: (Monad m) => Fold m Char b -> Stream m Char -> Stream m b
+words f m =  Stream.fromStreamD $ D.wordsBy isSpace f (Stream.toStreamD m)
```

Comment:
You should not use `StreamD` here. You can use `Stream.foldMany (Fold.takeEndBy_ isSpace)` instead.

Reply:
Maybe a good idea to keep it as it is so that we do not have to worry about perf comparison and write the `foldMany` snippet in a comment. We can replace it later if performance is found to be equivalent.

Reply:
Ideally StreamD should not lead into other modules. Stream is the aggregator. If required we can add `wordsBy` in Data.Stream and use this implementation there.

Reply:
As a quick solution, we can just leave this as it is and make a note here I guess.